app/models/concerns/profile_searching.rb

```diff
@@ -148,6 +149,33 @@ def filter_by_policy(_filter, _operator, policy_or_profile_id)
         conditions: "profiles.id IN (#{any_child_profile.to_sql})"
       }
     end
+
+    def filter_by_policy_and_profile_names(_filter, operator, value)
+      values = preprocess_parameters(operator, value)
+      cond = sanitize_sql_for_conditions(["policies.name #{operator} (?) OR profiles.name #{operator} (?)",
+                                          values, values])
+      search = left_outer_joins(:policy)
+      ids = search.where(cond)
+
+      { conditions: 'profiles.id IN (?)',
+        parameter: [ids.map(&:id)] }
```

Comment:
A subquery should be always faster than a map. Haven't tried, but `ids.select(:id)` might work, or the sub-optimal `ids.pluck(:id)` will work for sure. Having a `map` makes Rails to do the iteration which is way more expensive than just sending everything for PG.